### PR TITLE
Changed ajaxProcessUpdateImagePosition to be able to change the position of products images in multistore

### LINKED ISSUE #12832: Product image position not updatable in multishop [1.6.1.23]
Description : On a multistore PrestaShop, when you manage the images of a product in the BO, you can't change the images positions if one of them is not in the default shop

1. Install PrestaShop 1.6.23
2. Go to Preferences > General > enable multistore
3. Go to Advances parameters > Multistore
4. Create a new store group with all the basic settings
5. Create a new store with all the basic settings and attach it to the newly created store group
6. Go the a product with images, in the Images tab
7. Deselect the default shop for one of the images, wait for the green update confirmation popup in the top right corner
8. Try to change the position of one image, observe the loading image in the PrestaShop logo in the top left  corner (next to PrestaShop 1.6.1.23)

The problem cause this error in the console : 
prestashop 1 6 1 23 console error

When looking into the request generated by the drag and drop, we can see the response :
prestashop 1 6 1 23 request response

So, what happens ? Well, here is everything that happens from the moment you change a position : 
1. The function ajaxProcessUpdateImagePosition() is called in controllers/admin/AdminProductsController.php
2. This function then calls the Image class as seen in the screenshot below
prestashop 1 6 1 23 adminproductscontroller
3. The __construct() function in classes/Image.php then calls the __construct() function from classes/ObjectModel.php
prestashop 1 6 1 23 image class construct function
4. This function then select the id_shop from the current Context as seen in the screenshot below
prestashop 1 6 1 23 class objectmodel construct function
5. Then it calls the load function from Adapter/Adapter_EntityMapper.php from where it does the sql request to get the image which looks like this : 
SELECT *
FROM `ps_image` a
LEFT JOIN `ps_image_shop` `c` ON a.`id_image` = c.`id_image` AND c.`id_shop` = 1
WHERE (a.`id_image` = 1)

The problem is that the id_shop is the one from the current context, and if we deselect an image from the shop with the id 1 (for this exemple), it deletes in the database in the table ps_image_shop the corresponding line which this request is trying to get.

The issue comes from the fact that the Image class construct method does not take into account the id_shop of the image.

EDIT (in case someone might want to fix this issue on their shop) :

Simple Solution : Replace the inside of the foreach loop in the ajaxProcessUpdateImagePosition() in AdminProductsController with this code
                $res &= Db::getInstance()->execute('UPDATE ' . _DB_PREFIX_ . 'image SET position = ' . $position . ' WHERE id_image = ' . $id);

### PR DESCRIPTION
**Pull request by glictevo**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Related to the issue : https://github.com/PrestaShop/PrestaShop/issues/12832 for more details.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #12832 
| How to test?  | Follow instructions on how to reproduce on the issue

Comment : I know I've been told that it is not critical, but at least there'll be a simple solution if someone has this problem later with this version of PrestaShop, even if not actually pulled.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12847)

### CODE REVIEW

File: controllers/admin/AdminProductsController.php [hunk 1]
```diff
@@ -1477,9 +1477,7 @@ public function ajaxProcessUpdateImagePosition()
             $json = stripslashes($json);
             $images = json_decode($json, true);
             foreach ($images as $id => $position) {
-                $img = new Image((int) $id);
-                $img->position = (int) $position;
-                $res &= $img->update();
+                $res &= Db::getInstance()->execute('UPDATE ' . _DB_PREFIX_ . 'image SET position = ' . $position . ' WHERE id_image = ' . $id);
```
Comment (PierreRambaud) [inline controllers/admin/AdminProductsController.php:1480]: Is this problem still exists in the latest release?

Comment (glictevo) [reply controllers/admin/AdminProductsController.php:1480]: I have no idea, this was for 1.6, even though the bug was there in 1.7.6.1 (https://github.com/PrestaShop/PrestaShop/issues/12832#issuecomment-526241845)

I haven't been working on PrestaShop for over a year now so I'm totally out of the loop

Comment (PierreRambaud) [reply controllers/admin/AdminProductsController.php:1480]: I will try to investigate to see if the problem still exists in 1.7, and if your fix is the right one. Thanks for your quick reply

**Comment by PierreRambaud** [comment]
Hi @glictevo ,

Since the page is currently migrated in the next version, we decide to close this one. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**